wellcomeml/ml/bert_vectorizer.py

```diff
@@ -40,14 +40,15 @@ def load_transformed(cls, path):
         return np.load(path)
 
     def bert_embedding(self, x):
-        # Max sequence length is 512 for BERT
-        if len(x) > 512:
-            embedded_a = self.bert_embedding(x[:512])
-            embedded_b = self.bert_embedding(x[512:])
+        tokenized_x = self.tokenizer.tokenize(x)
+
+        # Max sequence length is 512 for BERT. 510 without CLS and SEP
+        if len(tokenized_x) > 510:
+            embedded_a = self.bert_embedding(" ".join(tokenized_x[:510]))
```

Comment:
I see, we are approximating by joining them with " ". It might miss some edge cases, but probably ok!

Reply:
whatever the reason for splitting them before, space should make them split again is my theory so its only safe to add spaces there